File: src/main/java/net/hectus/neobb/turn/default_game/block/TPowderSnow.java
package net.hectus.neobb.turn.default_game.block;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.ColdClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.inventory.ItemStack;

import java.util.List;

public class TPowderSnow extends BlockTurn implements BuffFunction, ColdClazz {
    public TPowderSnow(NeoPlayer player) { super(player); }
    public TPowderSnow(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public ItemStack item() {
        return new ItemStack(Material.POWDER_SNOW_BUCKET);
    }

    @Override
    public int cost() {
        return 5;
    }

    @Override
    public void apply() {
        player.nextPlayer().player.setFreezeTicks(6000); // 5 Minutes Max
        player.game.turnScheduler.runTaskLater("freeze", () -> player.game.eliminatePlayer(player.nextPlayer()), 3);
    }

    @Override
    public int maxAmount() { return 1; }

    @Override
    public List<Buff> buffs() { return List.of(); }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/block/TPowderSnow.kt
package net.hectus.neobb.modes.turn.default_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.game.util.ScheduleID
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.ColdClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.BuffFunction
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.Material
import org.bukkit.block.Block
import org.bukkit.inventory.ItemStack

class TPowderSnow(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), BuffFunction, ColdClazz {
    override val cost: Int = 5
    override val maxAmount: Int = 1

    override fun item(): ItemStack = ItemStack(Material.POWDER_SNOW_BUCKET)

    override fun apply() {
        player!!.nextPlayer().player.freezeTicks = 6000
        player.game.turnScheduler.runTaskLater(ScheduleID.FREEZE, 3) {
            player.game.eliminate(player.nextPlayer())
        }
    }

    override fun buffs(): List<Buff<*>> = emptyList()
}
